### PR TITLE
feat: toggle highlight using ctrl/cmd + shift + h

### PR DESCRIPTION
**Pull request by MayurSMahajan**
Solves: #327

### CODE REVIEW

**Comment by LucasXu0** [comment]
The logic does not seem correct. We should cancel the highlight color if it is not equal to the `defaultHighlightColor`.

**Comment by MayurSMahajan** [comment]
@Xazin yes you can help me, review the PR and let me know of any improvements

File: lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  EditorState editorState, {
+  required ToggleColorsStyle style,
+}) {
+  if (PlatformExtension.isMobile) {
+    assert(false, 'toggle highlight is not supported on mobile platform.');
+    return KeyEventResult.ignored;
+  }
+
+  final selection = editorState.selection;
+  if (selection == null) {
```
Comment (Xazin) [inline lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart:49]:
```suggestion
  if (selection == null || selection!.isCollapsed) {
```
We don't need to do anything if it's collapsed right?

File: lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    return KeyEventResult.ignored;
+  }
+
+  //check if already highlighted
+  final nodes = editorState.getNodesInSelection(selection);
+  final isHighlight = nodes.allSatisfyInSelection(selection, (delta) {
```
Comment (Xazin) [inline lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart:55]:
```suggestion
  final isHighlighted = nodes.allSatisfyInSelection(selection, (delta) {
```

File: lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    return delta.everyAttributes(
+      (attributes) => attributes[AppFlowyRichTextKeys.highlightColor] != null,
```
Comment (Xazin) [inline lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart:57]: I believe what Lucas wanted you to do, is:

- If the highlight is already `== style.highlightColor` then do nothing
- If the highlight is another color then set to `style.highlightColor`
- If there is no highlight then set to `style.highlightColor`.

So this check needs to be changed.

File: lib/src/editor/block_component/standard_block_components.dart [hunk 4]
```diff
@@ -111,6 +111,7 @@ final List<CommandShortcutEvent> standardCommandShortcutEvents = [
   //
   toggleTodoListCommand,
   ...toggleMarkdownCommands,
+  ...toggleColorCommands(),
```
Comment (Xazin) [inline lib/src/editor/block_component/standard_block_components.dart:114]: Here we need to provide the highlight color which is the `default highlight color`. Part of Lucas' comment.

File: lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart (hunk 3 above)
Comment (MayurSMahajan) [reply lib/src/editor/editor_component/service/shortcuts/command_shortcut_events/toggle_colors_command.dart:57]: cc @LucasXu0

**Review by Xazin — APPROVED** [review]

**Pull request merged**